src/ProbabilisticProgramming/Monad.dfy

```diff
@@ -7,6 +7,54 @@ module Monad {
   import Rand
   import Measures
 
+  export Spec
+    provides
+      Hurd,
+      Return,
+      Bind,
+      Coin,
+      Eval,
+      Rand,
+      Map,
+      Result,
+      Result.RestIn,
+      Result.In,
+      Result.Satisfies,
+      Result.Bind,
+      Result.Map,
+      Result.Equals,
+      Result.IsConverging,
+      Result.IsDiverging,
+      Result.ValueOf,
+      Result.RestOf,
```

Comment:
Why don't we reveal all of `Result`? The important encapsulation is the Hurd type, isn't it?

Reply:
I guess hiding the `Rest` of a `Result` would make sense, so making `Result` opaque is probably a good idea after all.